response_test.go

```diff
@@ -525,6 +526,314 @@ func TestMarshalISO8601TimePointer(t *testing.T) {
+		Decimal:       sql.NullInt32{Valid: true},
+		Fractional:    sql.NullInt64{Valid: true},
+		ComputedAt:    sql.NullTime{Valid: true},
+		ComputedAtISO: sql.NullTime{Valid: true},
+	}
+
+	out := bytes.NewBuffer(nil)
+	if err := MarshalPayload(out, pi); err != nil {
+		t.Fatal(err)
+	}
+
+	var jsonData map[string]interface{}
+
+	if err := json.Unmarshal(out.Bytes(), &jsonData); err != nil {
+		t.Fatal(err)
+	}
+	data := jsonData["data"].(map[string]interface{})
+
+	if _, ok := data["id"]; ok {
+		t.Fatal("Was expecting data.id to be omitted")
+	}
+
+	if data["type"] != "null-string-id" {
+		t.Fatalf("Error marshalling type")
+	}
+
+	attributes := data["attributes"].(map[string]interface{})
+
+	if attributes["periodic"] != false {
+		t.Fatalf("Error marshalling to sql.NullBool: %v", attributes["periodic"])
+	}
+
+	if attributes["name"] != "" {
+		t.Fatal("Error marshalling to sql.NullString")
+	}
+
+	if attributes["value"] != 0.0 {
+		t.Fatal("Error marshalling to sql.NullFloat64")
+	}
+
+	if attributes["decimal"] != 0.0 {
+		t.Fatalf("Error marshalling to sql.NullInt32")
+	}
+
+	if attributes["fractional"] != 0.0 {
+		t.Fatalf("Error marshalling to sql.NullInt64")
+	}
+
+	if _, ok := attributes["computed_at"]; ok {
+		t.Fatal("Was expecting data.attributes.computed_at to be omitted")
+	}
+
+	if _, ok := attributes["computed_at_iso"]; ok {
+		t.Fatal("Was expecting data.attributes.computed_at_iso to be omitted")
+	}
+}
+
+func TestMarshalStructNullStringID(t *testing.T) {
+	pi := &NullStringID{
+		ID: sql.NullString{
+			String: "314",
+			Valid:  true,
+		},
+		Periodic: sql.NullBool{
+			Bool:  false,
+			Valid: true,
+		},
+		Name: sql.NullString{
+			String: "Pi",
+			Valid:  true,
+		},
+		Value: sql.NullFloat64{
+			Float64: 3.1415926535897932,
+			Valid:   true,
+		},
+		Decimal: sql.NullInt32{
+			Int32: 3,
+			Valid: true,
+		},
+		Fractional: sql.NullInt64{
+			Int64: 1415926535897932,
+			Valid: true,
+		},
+		ComputedAt: sql.NullTime{
+			Time:  time.Date(2021, 3, 14, 15, 0, 0, 0, time.UTC),
+			Valid: true,
+		},
+		ComputedAtISO: sql.NullTime{
+			Time:  time.Date(2021, 3, 14, 15, 0, 0, 0, time.UTC),
+			Valid: true,
+		},
+	}
+
+	out := bytes.NewBuffer(nil)
+	if err := MarshalPayload(out, pi); err != nil {
+		t.Fatal(err)
+	}
+
+	resp := new(OnePayload)
+	if err := json.Unmarshal(out.Bytes(), resp); err != nil {
+		t.Fatal(err)
+	}
+
+	data := resp.Data
+
+	if data.ID != "314" {
+		t.Fatal("Error marshalling id")
+	}
+
+	if data.Type != "null-string-id" {
+		t.Fatal("Error marshalling type")
+	}
+
+	if data.Attributes["periodic"] != false {
+		t.Fatal("Error marshalling to sql.NullBool")
+	}
+
+	if data.Attributes["name"] != "Pi" {
+		t.Fatal("Error marshalling to sql.NullString")
+	}
+
+	if data.Attributes["value"] != 3.1415926535897932 {
+		t.Fatal("Error marshalling to sql.NullFloat64")
+	}
+
+	if data.Attributes["decimal"] != 3.0 {
+		t.Fatalf("Error marshalling to sql.NullInt32")
+	}
+
```

Comment:
Unlike in the previous test, I'm not checking the expected values of `data.Attributes["fractional"]` and `data.Attributes["computed_at"]`, since what we're getting doesn't make much sense to me.

I would expect `data.Attributes["fractional"]` to be `1415926535897932`, yet we get `1.415926535897932e+15`.
The same goes for `data.Attributes["computed_at"]`, where we get `1.615734e+09` instead of `1615734000`.

I'm not sure if this is expected, but it doesn't feel right. I also wonder if this is somehow related to #202, where we get a different `float64` when converting values back.